File: app/src/main/java/com/example/mrakopediareader/categorieslist/CategoriesByPage.java
package com.example.mrakopediareader.categorieslist;

import com.example.mrakopediareader.R;
import com.example.mrakopediareader.api.Category;

import java.util.ArrayList;

import io.reactivex.rxjava3.core.Observable;

public class CategoriesByPage extends CategoriesList {
    @Override
    protected Observable<ArrayList<Category>> getCategories() {
        assert api != null;
        final String pageTitle = getIntent()
                .getStringExtra(getResources().getString(R.string.pass_page_title));
        return api.getCategoryByPage(pageTitle);
    }
}

File: app/src/main/java/com/example/mrakopediareader/categorieslist/CategoriesByPage.kt
package com.example.mrakopediareader.categorieslist

import com.example.mrakopediareader.R
import com.example.mrakopediareader.api.dto.Category
import io.reactivex.rxjava3.core.Observable

class CategoriesByPage : CategoriesList() {
    override fun getCategories(): Observable<List<Category>> {
        val pageTitle = intent.getStringExtra(resources.getString(R.string.pass_page_title))
        return if (pageTitle == null) {
            Observable.error(Throwable("No page title"))
        } else {
            api.getCategoryByPage(pageTitle)
        }
    }
}